alns/accept/SimulatedAnnealing.py

```diff
@@ -154,12 +163,16 @@ def autofit(
         if not (0 < accept_prob < 1):
             raise ValueError("accept_prob outside (0, 1) not understood.")
 
-        if num_iters < 0:
-            raise ValueError("Negative number of iterations not understood.")
+        if num_iters <= 0:
+            raise ValueError("number of iterations <= 0 not understood.")
 
         start_temp = -worse * init_obj / np.log(accept_prob)
-        step = (1 / start_temp) ** (1 / num_iters)
+
+        if method == "exponential":
+            step = (1 / start_temp) ** (1 / num_iters)
+        else:
+            step = (start_temp - 1) / num_iters
```

Comment:
Issue: removing the `method.lower()` from here does not lead to the correct behavior.  E.g., `method == "EXPONENTIAL"` will be interpreted as linear step.

Reply:
One way to fix this is to check for `method.lower()`.

Reply:
Yeah no, I got this wrong. Lemme fix.